File: libraries/uava/src/test/java/com/example/uava/PreconditionsTest.java
package com.example.uava;

import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class PreconditionsTest {
	@Test
	public void shouldThrowExceptionAndReturnProvidedMessageWhenValueIsNull() {
		try {
			Preconditions.checkNotNull(null, "the message");
		} catch (NullPointerException ex) {
			Assert.assertEquals(ex.getMessage(), "the message");
		}
	}

	@Test
	public void shouldReturnSameValueWhenNotNull() {
		String data1 = "test1";
		data1 = Preconditions.checkNotNull(data1);

		String data2 = "test2";
		data2 = Preconditions.checkNotNull(data2, "the message");

		Assert.assertEquals(data1, "test1");
		Assert.assertEquals(data2, "test2");
	}

	@Test
	public void shouldThrowExceptionWhenArgumentDoesNotSatisfyExpression() {
		try {
			Preconditions.checkArgument(false, "error message");
		} catch (IllegalArgumentException ex) {
			Assert.assertEquals(ex.getMessage(), "error message");
		}
	}

	@Test
	public void shouldNotThrowExceptionWhenArgumentSatisfyExpression() {
		Preconditions.checkArgument(true, "message");
	}

}

File: app/src/test/java/com/example/utils/PreconditionsTest.kt
package com.example.utils

import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

@RunWith(JUnit4::class)
class PreconditionsTest {

    @Test
    fun shouldThrowExceptionAndReturnProvidedMessageWhenValueIsNull() {
        try {
            Preconditions.checkNotNull(null, "the message")
        } catch (ex: NullPointerException) {
            assertEquals(ex.message, "the message")
        }
    }

    @Test
    fun shouldReturnSameValueWhenNotNull() {
        var data1 = "test1"
        data1 = Preconditions.checkNotNull(data1)

        var data2 = "test2"
        data2 = Preconditions.checkNotNull(data2, "the message")

        assertEquals(data1, "test1")
        assertEquals(data2, "test2")
    }

    @Test
    fun shouldThrowExceptionWhenArgumentDoesNotSatisfyExpression() {
        try {
            Preconditions.checkArgument(false, "error message")
        } catch (ex: IllegalArgumentException) {
            assertEquals(ex.message, "error message")
        }
    }

    @Test
    fun shouldNotThrowExceptionWhenArgumentSatisfyExpression() {
        Preconditions.checkArgument(true, "message")
    }

}